### PR TITLE
Allow enabling per-jail SYSV IPC namespaces in FreeBSD 11

### PR DESCRIPTION
**Pull request by asomers**
Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Supply documentation according to [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)
- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)
- [x] Only open the PR against the `develop` branch.

In FreeBSD 11.0 and later, SYSV message, semaphore, and shared memory
primitives can have per-jail namespaces.  This is controlled by the
"sysvshm", "sysvmsg", and "sysvsem" jail properties.  This commit adds
support for those properties to iocage.  The default is to enable sysv
IPC in jails, with per-jail namespaces.

XXX this PR contains a bug!  If you create a jail on an older version of iocage-develop and then upgrade to this one, the jail cannot be started, because the new properties won't exist in the ucl config file.  In PR #370 I deal with that by adding the `__get_sysv_prop` function, but that's not possible on develop because __get_jail_prop calls __die if the property isn't found.  Another option would be to upgrade the ucl config file whenever iocage itself is updated, but there doesn't seem to be a mechanism for that.  @skarekrow can you suggest a solution?  Please don't merge this PR without one.

### CODE REVIEW

**Comment by skarekrow** [comment]
Hmm, I'll think on this and see what I come up with. Shouldn't be too bad to solve. Thanks for the PR's!

**Comment by bartekrutkowski** [comment]
Has anyone noticed that setting the allow_ properties (including allow_sysvipc) fails completely on iocage-devel? I've reviewed this code, but I don't think it includes a fix for that issue, and if it doesn't then wouldn't it mean it can't work at all?

**Comment by skarekrow** [comment]
Last I used devel @bartekrutkowski (it's been a while granted!) that functionality worked for me. Can you show any examples that are failing?

File: lib/ioc-rc [hunk 1]
```diff
@@ -270,12 +270,22 @@ __vnet_start () {
         ${_dataset})"
     _tmpfs="allow.mount.tmpfs=$(__get_jail_prop allow_mount_tmpfs ${_fulluuid} \
         ${_dataset})"
+     _sysvmsg="sysvmsg=$(__get_jail_prop sysvmsg ${_fulluuid} ${_dataset})"
+     _sysvsem="sysvsem=$(__get_jail_prop sysvsem ${_fulluuid} ${_dataset})"
+     _sysvshm="sysvshm=$(__get_jail_prop sysvshm ${_fulluuid} ${_dataset})"
```
Comment (skarekrow) [inline lib/ioc-rc:275]: @asomers I don't know why this didn't occur to me before, but let's move 273-275 into an else statement for 282. That should workaround this issue I believe.

File: lib/ioc-rc [hunk 2]
```diff
@@ -340,6 +353,9 @@ __legacy_start () {
         ${_dataset})"
     _tmpfs="allow.mount.tmpfs=$(__get_jail_prop allow_mount_tmpfs \
         "${_fulluuid}" ${_dataset})"
+     _sysvmsg="sysvmsg=$(__get_jail_prop sysvmsg ${_fulluuid} ${_dataset})"
+     _sysvsem="sysvsem=$(__get_jail_prop sysvsem ${_fulluuid} ${_dataset})"
+     _sysvshm="sysvshm=$(__get_jail_prop sysvshm ${_fulluuid} ${_dataset})"
```
Comment (skarekrow) [inline lib/ioc-rc:358]: Same suggestion as above @asomers. This can likely be an else statement for 370.

**Comment by bartekrutkowski** [comment]
@skarekrow The example is very simple, like that:
```
# iocage set allow_sysvipc="1" jailname
entered allow case
finished allow case
*** mountpoint /iocage/jails/9361d090-c6d7-11e6-9d6e-00163e001390
*** _pname allow.sysvipc
*** _pval 1
Failed to find destination node: allow.sysvipc
Error: Failed to apply the set operation.
```

**Comment by bartekrutkowski** [comment]
@skarekrow I think I've found it: the _pname transformation that's happening in `ioc-common` around line 320 works for the `jail` command parameter set, but fails for `__ucl_set` due to the dots in the parameter name. I've replaced it with a bit of code looking like that:
```
            if [ ! -z "${_jid}" ] ; then
                # These actually still want one _ instead of all periods.
                if [ "${_pname}" = "allow_raw_sockets" \
                    -o "${_pname}" = "allow_socket_af" \
                    -o "${_pname}" = "allow_set_hostname" ] ; then
                    _jail_pname="$(echo ${_pname} | sed 's/_/./')"
                else
                    _jail_pname="$(echo ${_pname} | sed 's/_/./g')"
                fi

                jail -m "jid=${_jid}" "${_jail_pname}=${_pval}"
            fi
```
I am testing it now in iocell and it does seem to be working. After some more testing I will commit that to the develop branch of iocell, feel free to grab the code and fix iocage too.

**Comment by bartekrutkowski** [comment]
@skarekrow See the fix in https://github.com/bartekrutkowski/iocell/issues/5

File: lib/ioc-rc (hunk 1 above)
Comment (asomers) [reply lib/ioc-rc:275]: @skarekrow we could certainly move 273-275 as you suggest.  But that wouldn't solve the upgrade issue I described in the PR.  Unless you think that these properties should be hard-coded to "new" and not be user-settable?

Comment (skarekrow) [reply lib/ioc-rc:275]: @asomers Oh man I was exhausted. You're totally right. Let's leave this one alone, I think it's fine as is. I'll merge it as soon as we figure out the develop one.

Comment (skarekrow) [reply lib/ioc-rc:275]: And of course I didn't even notice the branch I was on. What a week.

**Comment by skarekrow** [comment]
Appreciate it :)

**Comment by skarekrow** [comment]
Alright @asomers got the migration nailed. Will commit it along with your stuff :)

**Pull request merged**